tests/test_authentication.py

```diff
@@ -0,0 +1,50 @@
+import os
+import unittest
+from contextlib import contextmanager
+from unittest.mock import patch
+
+from lsst.daf.butler.remote_butler._authentication import (
```

Comment:
Given the `_` in the import path here the implication is that some of those would be lifted to `lsst.daf.butler.remote_butler` namespace. If the two package globals are private (they are not in `__all__`) then maybe the imports should be:

```python
from lsst.daf.butler.remote_butler import (
    get_authentication_headers,
    get_authentication_token_from_environment,
)
from lsst.daf.butler.remote_butler._authentication import (
    EXPLICIT_BUTLER_ACCESS_TOKEN_ENVIRONMENT_KEY,
    RSP_JUPYTER_ACCESS_TOKEN_ENVIRONMENT_KEY,
)
```
(and maybe even with leading underscores if they are really private). If they aren't private they will need docstrings.